### PR TITLE
Adding code generation code and templates

### PR DESCRIPTION
**Pull request by MichaelClerx**
See #85

- Removed dependency on `weblab_cg`
- Copied in templates (including temporary ones for reference)
- Copied in weblab printer, and template processing code
- Copied in tests
- Updated HH model with more annotations (used in tests for code generation)
- Added local version of `_exp` hack for sympy

### CODE REVIEW

File: fc/templates/unused/TestClampingToInitialValue.pyx [hunk 1]
```diff
@@ -0,0 +1,135 @@
+# cython: profile=True
```
Comment (jonc125) [inline fc/templates/unused/TestClampingToInitialValue.pyx]: I suspect these were useful when you were first figuring out the template, but aren't really needed now. However, let's keep them in this PR and delete when we're a bit further forward with all the code generation features.

File: fc/code_generation.py [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+            'var_name': var_name,
+            'parameter_index': parameter_index,
+            'length': length,
+        })
+
+    # Create RHS equation information dicts
+    rhs_equations = []
+    for eq in model.get_equations_for(model.get_derivative_symbols()):
+        # TODO: Parameters should never appear as the left-hand side of an
+        # equation (cellmlmanip should already have filtered these out).
```
Comment (jonc125) [inline fc/code_generation.py]: This is probably not true now... (but probably shouldn't fix in this PR?)

File: fc/code_generation.py [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+    for i, parameter in enumerate(parameters):
+        symbol = model.get_symbol_by_ontology_term(*parameter)
+        parameter_info.append({
+            'index': i,
+            'annotation': parameter,
+            'var_name': symbol_name(symbol),
+            'initial_value': model.get_value(symbol),
+        })
+
+    # Create map of parameter symbols to their indices
```
Comment (jonc125) [inline fc/code_generation.py]: Merge this loop with the one above?

File: fc/templates/weblab_model.pyx [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+    #        Sundials.N_VDestroy_Serial(self._parameters)
+
+    def get_environment_map(self):
+        """
+        Get a map from ontology prefix to the environment containing model
+        variables annotated with that ontology.
+
+        See :meth:`fc.simulations.AbstractOdeModel.get_environment_map()`.
+        """
+        # TODO Some part of this might need to be generated
```
Comment (jonc125) [inline fc/templates/weblab_model.pyx]: Yeah, eventually it all should be, based on ontologies actually used in the protocol.

File: fc/code_generation.py (hunk 2 above)
Comment (MichaelClerx) [reply fc/code_generation.py]: Thanks. Made an issue for it https://github.com/ModellingWebLab/weblab-fc/issues/87

File: fc/templates/unused/TestClampingToInitialValue.pyx (hunk 1 above)
Comment (MichaelClerx) [reply fc/templates/unused/TestClampingToInitialValue.pyx]: Agreed! https://github.com/ModellingWebLab/weblab-fc/issues/88

**Pull request merged**